stock_helper/tests/test_location_get_closest_warehouse.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+
+from .common import StockHelperCommonCase
+
+
+class TestStockLocationGetClosestWarehouse(StockHelperCommonCase):
+    def test_get_closest_warehouse(self):
+        test_warehouse = self.env["stock.warehouse"].create(
+            {"name": "Test Warehouse", "code": "Test WH Code"}
+        )
+        test_warehouse.view_location_id.location_id = self.wh.lot_stock_id.id
+        location = test_warehouse.lot_stock_id
+
+        self.assertEqual(location.warehouse_id, self.wh)
```

Comment:
This fails but I don't have enough context knowledge to say what it is testing and how to fix it.

When running the tests locally the `location.warehouse_id` ID increased after each run, so this way of testing doesn't seem stable.

First run: `AssertionError: stock.warehouse(4,) != stock.warehouse(1,)`
Second run: `AssertionError: stock.warehouse(5,) != stock.warehouse(1,)`
etc.

Reply:
That's normal. The warehouse(1) is the one defined in demo data (from stock module). The warehouse `test_warehouse` is the one created in tests (and so the id is incrementing at each run).

That said, your test will never succeed as your structure looks like :

     [WAREHOUSE(1)]
                  ||
     [TEST WAREHOUSE]

As the parent of your view location of test warehouse is the stock location of the warehouse 1